src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java

```diff
@@ -0,0 +1,281 @@
+import io.sphere.sdk.cartdiscounts.commands.CartDiscountCreateCommand;
+import io.sphere.sdk.cartdiscounts.queries.CartDiscountQuery;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.queries.QueryPredicate;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+import static com.commercetools.sync.benchmark.BenchmarkUtils.CART_DISCOUNT_SYNC;
+import static com.commercetools.sync.benchmark.BenchmarkUtils.CREATES_AND_UPDATES;
+import static com.commercetools.sync.benchmark.BenchmarkUtils.CREATES_ONLY;
+import static com.commercetools.sync.benchmark.BenchmarkUtils.NUMBER_OF_RESOURCE_UNDER_TEST;
+import static com.commercetools.sync.benchmark.BenchmarkUtils.THRESHOLD_EXCEEDED_ERROR;
+import static com.commercetools.sync.benchmark.BenchmarkUtils.UPDATES_ONLY;
+import static com.commercetools.sync.benchmark.BenchmarkUtils.saveNewResult;
+import static com.commercetools.sync.commons.asserts.statistics.AssertionsForStatistics.assertThat;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_CART_PREDICATE_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_TARGET_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_VALUE_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.JANUARY_FROM;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.JANUARY_UNTIL;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.deleteCartDiscounts;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.getSortOrders;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_TARGET_CLIENT;
+import static com.commercetools.sync.integration.commons.utils.TypeITUtils.deleteTypes;
+import static com.commercetools.tests.utils.CompletionStageUtil.executeBlocking;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class CartDiscountSyncBenchmark {
+
+    private CartDiscountSyncOptions cartDiscountSyncOptions;
+    private List<String> errorCallBackMessages;
+    private List<String> warningCallBackMessages;
+    private List<Throwable> errorCallBackExceptions;
+
+    @BeforeAll
+    static void setup() {
+        deleteTypes(CTP_TARGET_CLIENT);
```

Comment:
why do we need to delete types?

Reply:
CartDiscounts could refer to the custom fields (Types).

Reply:
but it doesn't seem that you are using custom fields for the cart discounts you are creating in the benchmarks.

Reply:
2383e8f